social/src/routes/twitter/tweet.ts

```diff
@@ -16,33 +12,34 @@ const consumerSecret = process.env.TWITTER_CONSUMER_SECRET!;
 router.post(
   '/api/social/twitter/statuses/update',
   requireAuth,
-  [body('status').not().isEmpty().withMessage('Please provide a status')],
+  [
+    body('status').not().isEmpty().withMessage('Please provide a status'),
+    body('twitterUserId')
+      .notEmpty()
+      .isNumeric()
+      .withMessage('Please provide a valid Twitter user ID'),
+  ],
   validateRequest,
   async (req: Request, res: Response) => {
-    const { status } = req.body;
-    const tokens = await UserController.getUserTwitterTokens(
-      req.currentUser!.userId
+    const { status, twitterUserId } = req.body;
+    const {
+      oauthAccessToken,
+      oauthAccessTokenSecret,
+    } = await getTwitterAccountTokens(
+      req.currentUser!.userId,
+      String(twitterUserId)
     );
-
-    if (tokens && tokens.oauthAccessToken && tokens.oauthAccessTokenSecret) {
-      const { oauthAccessToken, oauthAccessTokenSecret } = tokens;
-      const T = new twit({
-        consumer_key: consumerKey,
-        consumer_secret: consumerSecret,
-        access_token: oauthAccessToken,
-        access_token_secret: oauthAccessTokenSecret,
-      });
-
-      try {
-        await T.post('statuses/update', { status: status });
-        res.sendStatus(204);
-      } catch (err) {
-        await UserController.deleteUserTwitterTokens(req.currentUser!.userId);
-        throw new FailedConnectionError();
-      }
-    } else {
-      // no account connected;
-      throw new ForbiddenError();
+    const T = new twit({
```

Comment:
`T` is used in the package documentation so that's the reason I used it like this. But I agree, `twitterClient` makes this more readable for anyone first seeing this code. Fixed